src/api/registerApplicationResourceResolver.ts

```diff
@@ -17,6 +19,10 @@ export function registerApplicationResourceResolver(id: string, resolver: AppRes
     applicationResourceResolvers[id] = resolver;
     ext.emitters.onDidRegisterResolver.fire(resolver);
 
+    const compat = new CompatibleBranchDataProvider(resolver, 'azureResourceGroups.loadMore' /** TODO: what is the correct value for this? */);
+
+    ext.v2.api.registerApplicationResourceBranchDataProvider(id, compat as unknown as BranchDataProvider<ApplicationResource, ResourceModelBase>);
```

Comment:
Can this use `CompatibleBranchDataItem` instead of `ResourceModelBase`?
```suggestion
    ext.v2.api.registerApplicationResourceBranchDataProvider(id, compat as unknown as BranchDataProvider<ApplicationResource, CompatibleBranchDataItem>);
```

Reply:
We don't need that cast anymore

Reply:
I put the `never` in on `getParent?` to denote that users of the API do not need to (and should not) implement it--since RGs can do it all. I think I'd rather have the cast here than lose that.